src/core/listener.c

```diff
@@ -439,6 +457,45 @@ QuicListenerIndicateDispatchEvent(
             Event);
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicListenerIndicateStopCompleteEvent(
+    _In_ QUIC_LISTENER* Listener
+    )
+{
+    QUIC_LISTENER_EVENT Event;
+    Event.Type = QUIC_LISTENER_EVENT_STOP_COMPLETE;
+    Event.STOP_COMPLETE.AppCloseInProgress = Listener->AppClosed;
+
+    //
+    // Take an internal cleanup reference to prevent an inline ListenerClose
+    // freeing the listener from under us.
+    //
+    QuicListenerInternalReference(Listener);
+
+    QuicListenerAttachSilo(Listener);
+
+    QuicTraceLogVerbose(
+        ListenerIndicateStopComplete,
+        "[list][%p] Indicating STOP_COMPLETE",
+        Listener);
+
+    Listener->StopCompleteThreadID = CxPlatCurThreadID();
+    (void)QuicListenerIndicateEvent(Listener, &Event);
+    Listener->StopCompleteThreadID = 0;
+
+    QuicListenerDetachSilo();
+
+    //
+    // If !Listener->NeedsCleanup, then another thread is waiting on this event
```

Comment:
This comment is referencing behavior that is no longer true/not near this comment anymore. Please update/move or remove.